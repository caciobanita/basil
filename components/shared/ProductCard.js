import { Component } from 'react';
import Link from 'next/link';
import getConfig from 'next/config';

const { publicRuntimeConfig: config } = getConfig();

export default (props) => {
  let item = props.item;

  return (
    <div className="product-card">
      <Link href={`/content?slug=${item.url_key}`} as={`/${item.url_key}`}>
        <a className="product-card__content" key={item.id}>
          <img src={config.storeUrl + 'media/catalog/product' + item.small_image} />
          <h4 className="pt-2" dangerouslySetInnerHTML={{__html: item.name}}></h4>
          <div className="py-2">${item.price.regularPrice.amount.value}</div>
        </a>
      </Link>
    </div>
  );
}
