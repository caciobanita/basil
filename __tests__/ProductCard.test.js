import React from 'react';
import Renderer from 'react-test-renderer';
import ProductCard from '../components/shared/ProductCard';
import { shallow, mount, render } from 'enzyme';


const item = {
  "id": 68,
  "name": "Chaz Kangeroo Hoodie",
  "small_image": "\/m\/h\/mh01-gray_main_1.jpg",
  "url_key": "chaz-kangeroo-hoodie",
  "description": "<p>Ideal for cold-weather training or work outdoors, the Chaz Hoodie promises superior warmth with every wear. Thick material blocks out the wind as ribbed cuffs and bottom band seal in body heat.<\/p>\n<p>&bull; Two-tone gray heather hoodie.<br \/>&bull; Drawstring-adjustable hood. <br \/>&bull; Machine wash\/dry.<\/p>",
  "media_gallery_entries": [
    {
      "label": "",
      "file": "\/m\/h\/mh01-gray_main_1.jpg"
    },
    {
      "label": "",
      "file": "\/m\/h\/mh01-gray_alt1_1.jpg"
    },
    {
      "label": "",
      "file": "\/m\/h\/mh01-gray_back_1.jpg"
    }
  ],
  "price": {
    "regularPrice": {
      "amount": {
        "value": 52,
        "currency": "USD"
      }
    }
  }
};

describe('Test product card', () => {
  it('Renders correctly', () => {
    const renderer = Renderer.create(
      <ProductCard item={item} />
    );
    const tree = renderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test product card name', () => {
  const wrapper = shallow(<ProductCard item={item} />);

  expect(wrapper.find('h4')).toHaveLength(1);
  expect(wrapper.find('h4').html()).toEqual('<h4 class="pt-2">Chaz Kangeroo Hoodie</h4>');
});
