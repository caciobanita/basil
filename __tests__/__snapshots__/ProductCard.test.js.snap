// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test product card Renders correctly 1`] = `
<div
  className="product-card"
>
  <a
    className="product-card__content"
    href="/chaz-kangeroo-hoodie"
    onClick={[Function]}
  >
    <img
      src="http://m2.localhost/media/catalog/product/m/h/mh01-gray_main_1.jpg"
    />
    <h4
      className="pt-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Chaz Kangeroo Hoodie",
        }
      }
    />
    <div
      className="py-2"
    >
      $
      52
    </div>
  </a>
</div>
`;
